### ISSUE TITLE
Need ALLOWED_TELEGRAM_GROUP_IDS in Variable

### ISSUE THREAD

**Issue by drbuilderpro** [n3d1117/chatgpt-telegram-bot#276]
Now Only "ALLOWED_TELEGRAM_USER_IDS" available in varaibles. By adding "ALLOWED_TELEGRAM_GROUP_IDS" need to allow all members of a specific group access bot. When enabled "ALLOWED_TELEGRAM_GROUP_IDS", bot should not work for "ALLOWED_TELEGRAM_USER_IDS" also in other chats. Even if a chat is not added in "ALLOWED_TELEGRAM_GROUP_IDS", if "ADMIN" bot should answer.

Can You Help Please

**Comment by AlexHTW** [comment]
Hey @drbuilderpro, I think most of want you need can be done with the current features. For this part...

> By adding "ALLOWED_TELEGRAM_GROUP_IDS" need to allow all members of a specific group access bot.
When enabled "ALLOWED_TELEGRAM_GROUP_IDS", bot should not work for "ALLOWED_TELEGRAM_USER_IDS" also in other chats.
Even if a chat is not added in "ALLOWED_TELEGRAM_GROUP_IDS", if "ADMIN" bot should answer.

... you could make use of the guest feature. 

If you are in a group with the bot (and the bot has permission to read the messages) everyone in the group is allowed to use the bot within the shared `GUEST_BUDGET` limit.
I would suggest also adding a `GROUP_TRIGGER_KEYWORD`.  For example your `.env` could look like this:

```.env
# access only in admin groups
ADMIN_USER_IDS=YOUR_ID
ALLOWED_TELEGRAM_USER_IDS=YOUR_ID
GUEST_BUDGET=10.0
BUDGET_PERIOD=daily
GROUP_TRIGGER_KEYWORD=gpt
```
All members of the groups with you and the bot can prompt the bot when starting their message with `gpt`. All guests together can use $10.0 a day.

> Even if a chat is added in "ALLOWED_TELEGRAM_GROUP_IDS", if USER is not added in "ALLOWED_TELEGRAM_USER_IDS" bot should not answer to them.

I don't really understand the connection to the part above, but for this case you can just set the GUEST_BUDGET to 0.

**Comment by drbuilderpro** [comment]
Thanks for the reply. i was unaware of the use of GUEST_BUDGET. thank for the explanation. let me check

**Comment by AlexHTW** [comment]
Happy to help, let me know if you have more questions regarding this topic. You can also check out the Budget Manual #184 for more detailed explanations.

**Comment by drbuilderpro** [comment]
Actually how to make the bot work in allowed groups only?

**Comment by AlexHTW** [comment]
You can't set specific groups in the .env. You have to invite your bot into a group and give him read (or admin) permissions. Then everyone in the group can interact with the bot, as long as you are in the .env as admin or user.

**Comment by drbuilderpro** [comment]
but i dont want other members use the bot. GUEST_BUDGET=0 , can make all other members restricted? why we dont have ALLOWED_GROUPS Variable? Can you look into my repo to guide me what is wrong? GUEST_BUDGET=0 , also not restricting my bot to others?

**Comment by AlexHTW** [comment]
> but i dont want other members use the bot. GUEST_BUDGET=0 , can make all other members restricted?

Yes, if you don't want any other people than the users in `ADMIN_USER_IDS` and `ALLOWED_TELEGRAM_USER_IDS` to have access to the bot then you can set GUEST_BUDGET=0.
Then you won't have to invite the bot to the group and all users can just use the bot with inline mode in any group they want.

> why we dont have ALLOWED_GROUPS Variable? 

Well, as you can see yourself in this disscussion the more options for permissions the bot offers the more complicated it gets and adding ALLOWED_GROUPS could lead to a lot of changes for the current logic. I feel like the current settings are a good compromise between allowing a variety of setups without being overly complicated. But if you or anyone else has a comprehensive solution for ALLOWED_GROUPS go for it :).

> GUEST_BUDGET=0 , also not restricting my bot to others?

I don't understand.
I guess you want users to be able to use the bot in a group, but only in the group and nowhere else but other members of the group should not be able to use it. To be honest I would just create a new group with the people you want to allow the bot and allow them guest usage. 
Or you add them as users with very low budgets and accept that they might use the bot privately a little (you can always ask them personally not to and remove their access if you see they do it anyway).

> Can you look into my repo to guide me what is wrong? 

It looks like you just added a new environment variable but no logic that handles the access.
PR #147 by @dkvdm implemented the logic you are looking for, but they ultimately decided that the available settings are enough and closed it. You can check out the logic there if you want to implement it for the current version or maybe @dkvdm is interested in working on this with you...

Personally, I think this is a very edge case and not worth the effort

**Comment by drbuilderpro** [comment]
restriction should be restriction. if admin added, bot answer only to admin. if allowed users there, bot should only answer to them... OMG, this bot missing important things...😒

**Comment by AlexHTW** [comment]
> But My question is why the bot still answer if GUEST_BUDGET=0 ??? it shouldn't be like that right?

No it shouldn't be like that. Maybe you didn't set anything for `ALLOWED_TELEGRAM_USER_IDS` set, so it defaults to `*` and everybody is allowed. You should put something else that is not `*`, for example your ID.

> I want this bot to be restricted SO THAT IT WORKS only to me. 

Good, that is easily doable with the current settings, just set your ID as admin and user.

> You may not have any logic, but i do. 

I was referring to your repository that you asked me to look at, you just added an environment variable but the code is not doing anything with it.

> and my bot also not answering with inline query, when enabled trigger?

for inline query you use `@name_of_your_bot`, trigger only applies if you want the bot to answer directly to your messages in a group when the bot is also a member of the group.

> restriction should be restriction. if admin added, bot answer only to admin. if allowed users there, bot should only answer to them... OMG, this bot missing important things...

If you didn't set any `ALLOWED_TELEGRAM_USER_IDS` then it defaults to `*` which allows everyone. I understand that may be not what you expect but the readme and the budget manual clearly explain that.


So to summarize. For you I think it's best to use inline query.
Your `-env` should look like this:

```.env
# access only in admin groups
ADMIN_USER_IDS=YOUR_ID
ALLOWED_TELEGRAM_USER_IDS=YOUR_ID
GUEST_BUDGET=0
```

Then you can use the bot in any chat or group by typing `@name_of_your_bot`, wait for it to load, then type your question and press the pop up "ask ChatGPT" and button "answer with ChatGPT".

**Comment by drbuilderpro** [comment]
@AlexHTW

>I was referring to your repository that you asked me to look at, you just added an environment variable but the code is not doing anything with it.

My repo is private. not this forked one

**Comment by n3d1117** [comment]
Hi @drbuilderpro

> Sadly, inline query not answers, when enabled group trigger.

What do you mean? Could you elaborate
